backend/server/src/handler/role.rs

```diff
@@ -0,0 +1,48 @@
+use crate::models::app::AppState;
+use crate::models::auth::{AuthUser, OrganisationAdmin};
+use crate::models::error::ChaosError;
+use crate::models::role::{Role, RoleUpdate};
+use axum::extract::{Json, Path, State};
+use axum::http::StatusCode;
+use axum::response::IntoResponse;
+
+pub struct RoleHandler;
+
+impl RoleHandler {
+    pub async fn create(
+        State(state): State<AppState>,
+        Path(id): Path<i64>,
+        _admin: OrganisationAdmin,
+        Json(data): Json<RoleUpdate>,
+    ) -> Result<impl IntoResponse, ChaosError> {
+        Role::create(id, data, &state.db).await?;
+        Ok((StatusCode::OK, "Successfully created role"))
+    }
+    pub async fn get(
+        State(state): State<AppState>,
+        Path(id): Path<i32>,
+        _user: AuthUser,
+    ) -> Result<impl IntoResponse, ChaosError> {
+        let role = Role::get(id, &state.db).await?;
+        Ok((StatusCode::OK, Json(role)))
+    }
+
+    pub async fn delete(
+        State(state): State<AppState>,
+        Path(id): Path<i32>,
+        _admin: OrganisationAdmin,
+    ) -> Result<impl IntoResponse, ChaosError> {
+        Role::delete(id, &state.db).await?;
+        Ok((StatusCode::OK, "Successfully deleted role"))
+    }
+
+    pub async fn update(
+        State(state): State<AppState>,
+        Path(id): Path<i32>,
+        _admin: OrganisationAdmin,
```

Comment:
`OrganisationAdmin` won't work for `update()` and `delete()` as the only int in the path is the role's id. `OrganisationAdmin`'s `FromRequestParts` implementation expects there to be the id of the organisation.

Reply:
I would recommend creating a new struct called `RoleAdmin` and implementing `FromRequestParts` for it, using the role id to find the organisation it belongs to and thus if a user is an admin.

See `CampaignAdmin` implementation for more details.